### DISCUSSION TITLE
How to apply periodic boundary condition?

### DISCUSSION THREAD

**Discussion by arnab82** [block-hczhai/block2-preview#134] [Q&A] [answered]
Hi, I want to run dmrg calculation on a cyclic molecule, how do I enforce periodic boundary condition? Do you have any example of that?
Thanks in advance.

**Comment by hczhai** [comment, marked as answer]
What do you mean by "cyclic molecule"? Any molecule, [cyclic](https://en.wikipedia.org/wiki/Cyclic_compound) or not, is considered as a finite and isolated system and you do not need to add any periodic boundary conditions.

**Comment by arnab82** [comment]
I was referring to Polyaromatic hydrocarbon rings by "cyclic molecule". Thanks for clearing up the doubt. Thanks!

**Reply by hczhai** [reply]
To make it clear, I think you are not considering "polymers". For polymers, the periodic boundary condition is handled in the construction of second quantized Hamiltonian part. And ``block2`` will take whatever second quantized Hamiltonian (with pbc or without pbc) and do DMRG. For pbc or 2D system (with rings, for example), the accuracy may be limited at small bond dimension. In general you do not need to let ``block2`` know whether it is 1D, ring, obc, or pbc, as long as such information is correctly encoded in the Hamiltonian.

**Comment by arnab82** [comment]
Thank you for your previous clarification regarding pbc. I greatly appreciate your guidance.

I am unsure about how to formulate Hamiltonian for 2D system or pbc system, unlike I am using get_qc_mpo for normal cases and there is no keyword regarding pbc in that function.

Could you please provide further guidance or give any examples or references that shows how to formulate the Hamiltonian for PBC or 2D systems in block2?
Thank you!

**Reply by hczhai** [reply]
Please note that the input arguments of ``get_qc_mpo`` are "integrals" ``h1e`` and ``g2e``, which can be called the parameters in the Hamiltonian. The output of ``get_qc_mpo`` is the MPO.

1. **The (input) Hamiltonian parameters h1e and g2e includes the information for the topology of the system.** So if you change your system from 1D to 2D, or obc to pbc, the second quantized Hamiltonian will change, namely, ``h1e`` and ``g2e`` will change. This means the pbc or non-pbc or 2D or 1D information is already encoded in ``h1e`` and ``g2e``. There is no need to have additional keywords in ``get_qc_mpo`` to specify the topology of the system. ``block2`` is not responsible for computing ``h1e`` and ``g2e``.

2. **The (output) MPO includes the information for the topology of the tensor network (ansatz).** In ``block2``, "MPO" means a finite chain-like 1D tensor network. If you are expecting a "ring-like" tensor network or 2D tensor network (PEPO), those are not called "MPO". So ``get_qc_mpo`` will only produce a standard MPO. This is why there are no additional keywords in ``get_qc_mpo`` to specify the topology of the tensor network.

3. It is a common practice to use an MPO (1D finite tensor network) to solve non-finite-1D quantum chemistry systems. The accuracy of DMRG will be limited for such systems unless a sufficiently large bond dimension is used.

**Comment by arnab82** [comment]
Thanks a lot!
